src/pages/List/List.js

```diff
@@ -1,9 +1,87 @@
-import React from 'react';
-import { withRouter } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
+import axios from 'axios';
+import Result from './Result';
+import Filters from './Filters';
+import ListContents from './ListContents';
 
 function List() {
-  return;
+  const [data, setData] = useState({ productList: [], items: 10, preItems: 0 });
+
+  const getData = () => {
+    axios
+      .get('/data/List.json')
+      .then(res => {
+        let result = res.data.slice(data.preItems, data.items);
+        setData(prev => {
+          return { ...prev, productList: [...data.productList, ...result] };
+        });
+      })
+      .catch(error => {
+        alert(error);
+      });
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', infiniteScroll, true);
+    return window.removeEventListener('scroll', infiniteScroll, false);
+  }, []);
```

Comment:
- useEffect 에서 clean up 은 **함수를** return 하는 것이 syntax 입니다!
- 그리고 remove 하실때는, add 하실때와 인자를 동일하게 부여하셔야 합니다! 그래서 false -> true 로 수정해주세요!
- 사실 addEventListener 에서 useCapture 옵션은 필요 없을 것 같은데, 사용하시는 이유가 있나요?
```suggestion
  useEffect(() => {
    window.addEventListener('scroll', infiniteScroll);
    return () => window.removeEventListener('scroll', infiniteScroll);
  }, []);
```

Reply:
true를 적어주면 정상적으로 작동을 하지 않아서 적어뒀던 것 입니다!  clean up을 함수로 return하니 useCapture을 안써도 동작을 잘 하네요! 감사합니다 :)